packages/eslint-plugin/src/rules/no-tslint-comments.js

```diff
@@ -1,14 +1,19 @@
 // @ts-check
 const createRule = require('../utils/createRule');
 
+/**
+ * @typedef { import('@typescript-eslint/utils').TSESLint.RuleMetaDataDocs} RuleMetaDataDocs
+ * @typedef { import('@typescript-eslint/utils').TSESLint.RuleRecommendation} RuleRecommendation
+ */
+
 module.exports = createRule({
   name: 'no-tslint-comments',
   meta: {
     type: 'problem',
-    docs: {
+    docs: /** @type {RuleMetaDataDocs & { recommended: RuleRecommendation }} */ ({
```

Comment:
`meta.docs` type does (RuleMetaDataDocs) does not include `recommended` anymore. Added explicitly as [suggested in release note](https://typescript-eslint.io/blog/announcing-typescript-eslint-v8/#custom-rule-metadocs-types)

Reply:
Sorry for the drive by review, but I've been discussing this PR with @Hotell - How come you didn't update the createRule implementation to set the type? Then this verbosity and repetition in rule implementations is not needed

Reply:
One additional note, based on a very quick scan of the codebase, you do not appear to be leveraging this recommended property at all. It has no impact on linting, it is only valuable if you actually use it to e.g. derive a recommended config that users can extend from and/or if you are generating rule documentation from this data.

The alternative "fix" would therefore be to remove this property

Reply:
(This is part of the reason why we removed it from the base type in typescript-eslint)

Reply:
@JamesHenry Thanks, James! I appreciate the review. You are right, that we do not levering this property, I will remove this 👍🏻

Reply:
appreciate  another pair of eyes on this @JamesHenry , cheers !